### ISSUE TITLE
[themes] Explorer arrows do not update when using reload feature

### ISSUE THREAD

**Issue by JimiC** [Microsoft/vscode#71783]
- VSCode Version: 1.34.0-insider
- OS Version: All

Steps to Reproduce:

This issue is difficult to reproduce as it requires a special version of an icon theme extension that supports theme reloading (`_watch`).

**Comment by JimiC** [comment]
Being a maintainer of the `vscode-icons` extension, I stumbled on this issue while testing the implementation of this feature on our extension.

As I'm the author of the theme reloading feature, I pinpointed the cause of the bug. The reason for it is that on [reload](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/themes/browser/workbenchThemeService.ts#L219) we don't call the `onFileIconThemeChange` event, resulting in the [explorer tree](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/browser/parts/views/viewsViewlet.ts#L338) not getting notified.

The solution is to add a call to [doSetFileIconTheme](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/services/themes/browser/workbenchThemeService.ts#L514) on the `onApply` function.

Although much of the code execution of the `doSetFileIconTheme` is an overkill when we reload the icon theme, it has no side effects. 
Possible actions are:
1. Refactor `doSetFileIconTheme` and extract the calls to telemetry and event triggering to separate function and then use only the extracted function
2. Do nothing and just add the call to `doSetFileIconTheme` and don't care about the redundant code execution since it does not have any side effects.

I'm more then willing to provide a PR according the chosen action, although I suspect that you guys prefer action 2.

Awaiting your response.
